### PR TITLE
JAMES-3214 Remove MailRepository API from Server-Data api

### PR DESCRIPTION
**Pull request by chibenwa**
This is a little dependency cleanup following https://github.com/apache/james-project/pull/207#issuecomment-644317768

```
[INFO] --- maven-dependency-plugin:3.1.2:tree (default-cli) @ james-server-cli ---
[INFO] org.apache.james:james-server-cli:jar:3.6.0-SNAPSHOT
[INFO] +- org.apache.james:james-server-data-api:jar:3.6.0-SNAPSHOT:compile
[INFO] |  +- org.apache.james:james-core:jar:3.6.0-SNAPSHOT:compile
[INFO] |  |  +- com.sun.activation:javax.activation:jar:1.2.0:compile
[INFO] |  |  \- org.reactivestreams:reactive-streams:jar:1.0.3:compile
[INFO] |  \- org.apache.commons:commons-configuration2:jar:2.7:compile
[INFO] |     \- org.apache.commons:commons-text:jar:1.8:compile
[INFO] +- org.apache.james:james-server-mailbox-jmx:jar:3.6.0-SNAPSHOT:compile
[INFO] |  +- org.apache.james:apache-james-mailbox-api:jar:3.6.0-SNAPSHOT:compile
[INFO] |  |  +- org.apache.james:james-server-task-api:jar:3.6.0-SNAPSHOT:compile
[INFO] |  |  |  +- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[INFO] |  |  |  +- org.scala-lang:scala-library:jar:2.13.1:compile
[INFO] |  |  |  \- org.scala-lang.modules:scala-java8-compat_2.13:jar:0.9.0:compile
[INFO] |  |  \- com.ibm.icu:icu4j:jar:64.2:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.27:compile
[INFO] +- org.apache.james:james-server-util:jar:3.6.0-SNAPSHOT:compile
[INFO] |  +- org.apache.james:apache-mime4j-dom:jar:0.8.3:compile
[INFO] |  |  \- org.apache.james:apache-mime4j-core:jar:0.8.3:compile
[INFO] |  +- com.github.fge:throwing-lambdas:jar:0.5.0:compile
[INFO] |  +- com.github.steveash.guavate:guavate:jar:1.0.0:compile
[INFO] |  +- com.sun.mail:javax.mail:jar:1.6.2:compile
[INFO] |  +- commons-io:commons-io:jar:2.6:compile
[INFO] |  +- io.projectreactor:reactor-core:jar:3.3.4.RELEASE:compile
[INFO] |  +- javax.inject:javax.inject:jar:1:compile
[INFO] |  \- org.apache.commons:commons-lang3:jar:3.9:compile
[INFO] +- com.google.guava:guava:jar:25.1-jre:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:2.0.0:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.1.3:compile
[INFO] |  +- com.google.j2objc:j2objc-annotations:jar:1.1:compile
[INFO] |  \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.14:compile
[INFO] +- commons-cli:commons-cli:jar:1.4:compile
```

&&

```
130 benwa@horizon ..che/james-project/server/container/cli (git)-[mailrepository-regroup] % ls -lSah target/james-server-cli.lib                                                                           :(
total 27M
-rw-r--r--  1 benwa benwa  13M 16 août   2019 icu4j-64.2.jar
-rw-r--r--  1 benwa benwa 5,5M  9 déc.   2019 scala-library-2.13.1.jar
-rw-r--r--  1 benwa benwa 2,7M 27 juil.  2018 guava-25.1-jre.jar
-rw-r--r--  1 benwa benwa 1,5M 29 avril 08:06 reactor-core-3.3.4.RELEASE.jar
-rw-r--r--  1 benwa benwa 644K 15 août   2019 javax.mail-1.6.2.jar
-rw-r--r--  1 benwa benwa 626K 11 juil.  2019 scala-java8-compat_2.13-0.9.0.jar
-rw-r--r--  1 benwa benwa 608K 22 mai   09:31 commons-configuration2-2.7.jar
-rw-r--r--  1 benwa benwa 493K 13 août   2019 commons-lang3-3.9.jar
-rw-r--r--  1 benwa benwa 336K 27 juil.  2018 checker-qual-2.0.0.jar
-rw-r--r--  1 benwa benwa 323K 14 mars   2019 apache-mime4j-dom-0.8.3.jar
-rw-r--r--  1 benwa benwa 291K 16 juin  11:54 apache-james-mailbox-api-3.6.0-SNAPSHOT.jar
-rw-r--r--  1 benwa benwa 210K 20 déc.   2017 commons-io-2.6.jar
-rw-r--r--  1 benwa benwa 203K 26 mai   11:02 commons-text-1.8.jar
-rw-r--r--  1 benwa benwa 176K 16 juin  11:53 james-server-util-3.6.0-SNAPSHOT.jar
-rw-r--r--  1 benwa benwa 121K  8 nov.   2017 throwing-lambdas-0.5.0.jar
-rw-r--r--  1 benwa benwa 102K 14 mars   2019 apache-mime4j-core-0.8.3.jar
-rw-r--r--  1 benwa benwa  83K 16 juin  11:53 james-server-data-api-3.6.0-SNAPSHOT.jar
-rw-r--r--  1 benwa benwa  77K 12 juil.  2019 javax.activation-1.2.0.jar
-rw-r--r--  1 benwa benwa  53K  2 janv.  2018 commons-cli-1.4.jar
-rw-r--r--  1 benwa benwa  49K 16 juin  11:53 james-core-3.6.0-SNAPSHOT.jar
-rw-r--r--  1 benwa benwa  41K 15 août   2019 slf4j-api-1.7.27.jar
-rw-r--r--  1 benwa benwa  31K 16 juin  11:53 james-server-task-api-3.6.0-SNAPSHOT.jar
-rw-r--r--  1 benwa benwa  26K  3 avril  2019 javax.annotation-api-1.3.2.jar
-rw-r--r--  1 benwa benwa  24K 16 juin  11:56 james-server-mailbox-jmx-3.6.0-SNAPSHOT.jar
-rw-r--r--  1 benwa benwa  20K 29 mai    2018 jsr305-3.0.2.jar
-rw-r--r--  1 benwa benwa  14K  2 janv.  2018 error_prone_annotations-2.1.3.jar
-rw-r--r--  1 benwa benwa  13K  8 nov.   2017 guavate-1.0.0.jar
-rw-r--r--  1 benwa benwa  12K 27 sept.  2019 reactive-streams-1.0.3.jar
-rw-r--r--  1 benwa benwa 8,6K  2 janv.  2018 j2objc-annotations-1.1.jar
drwxr-xr-x  2 benwa benwa 4,0K 16 juin  11:57 ./
drwxr-xr-x 10 benwa benwa 4,0K 16 juin  11:57 ../
-rw-r--r--  1 benwa benwa 3,5K  2 janv.  2018 animal-sniffer-annotations-1.14.jar
-rw-r--r--  1 benwa benwa 2,5K  8 nov.   2017 javax.inject-1.jar
```

### CODE REVIEW

File: server/data/data-api/pom.xml [hunk 1]
```diff
@@ -39,25 +39,18 @@
...
         <dependency>
             <groupId>${james.groupId}</groupId>
-            <artifactId>james-server-mailrepository-api</artifactId>
+            <artifactId>testing-base</artifactId>
+            <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>${james.groupId}</groupId>
-            <artifactId>testing-base</artifactId>
+            <groupId>io.projectreactor</groupId>
```
Comment (ieugen) [inline server/data/data-api/pom.xml:46]: Aren't these interfaces already part of Java 9? 
https://www.reactive-streams.org/ . 

We could drop the dependency? 
I just skimmed over the docs, but that is my quick assessment.

Comment (chibenwa) [reply server/data/data-api/pom.xml:46]: Reactive streams might be but reactor definitly is not.

**Comment by chibenwa** [comment]
I propose to use JAMES-3214 Remove test only libraries from being pulled as transitive dependencies of production artifacts

as this PR had been opened in that context and adresses some of the concerns mentionned in that issue.

I propose to add the prefix upon merge.

File: server/data/data-memory/pom.xml [hunk 2]
```diff
@@ -76,7 +76,6 @@
         <dependency>
             <groupId>${james.groupId}</groupId>
             <artifactId>james-server-mailrepository-memory</artifactId>
-            <scope>test</scope>
```
Comment (rouazana) [inline server/data/data-memory/pom.xml:79]: why?

Comment (chibenwa) [reply server/data/data-memory/pom.xml:79]: Intermediate implementation stage...

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**